Add Rubocop

peterdenham opened the pull request:
Summary of changes:
- Added rubocop gem and dependencies
- Ran rubocop autocorrect for some quick fixes
- Added rubocop config file - disabling any currently failing cops to have a passing rubocop. These cops can be addressed in further PR's.

Bonus points: Add rubocop to github actions - https://github.com/andrewmcodes/rubocop-linter-action

Prime: @jasonswett

peterdenham committed:
aws wait

peterdenham committed:
rubocop first pass

peterdenham committed:
Revert "aws wait"

This reverts commit 44e7150d

peterdenham committed:
rubocop config - disable initial errors

jasonswett commented:
This all looks good with one exception: is it possible to use the `# frozen_string_literal: true` thing instead of `.freeze` individually on each string?

peterdenham committed:
move offences to rubocop todo

peterdenham committed:
Prefer frozen string literal magic comments

peterdenham committed:
Merge branch 'master' into feature/rubocop

# Conflicts:
#	lib/exosuit/help.rb

peterdenham commented:
Ah yes, I guess I was running an older version of ruby*. Updated to use magic comments and moved breaking cops to the todo rubocop file.

\* can we define the ruby version and gemset for the project too?

jasonswett commented:
> can we define the ruby version and gemset for the project too?

Seems like a good idea, yes

peterdenham committed:
Merge branch 'master' into feature/rubocop

# Conflicts:
#	Gemfile
#	lib/exosuit.rb
